libs/parcel-reporter-zephyr/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import { Reporter } from '@parcel/plugin';
+import { zeBuildDashData, ZephyrEngine } from 'zephyr-agent';
+import { getAssetsMap, ParcelOutputAsset } from './lib/get-assets-map';
+import * as path from 'path';
+
+// Create the engine and assets map outside the reporter function
+// so they persist between calls
+const { zephyr_engine_defer, zephyr_defer_create } = ZephyrEngine.defer_create();
+const assets = new Map<string, ParcelOutputAsset>();
+
+export default new Reporter({
+  report: async ({ event, options }) => {
+    const projectRoot = options.projectRoot;
+
+    if (event.type === 'buildStart') {
+      zephyr_defer_create({
+        builder: 'parcel',
+        context: projectRoot,
+      });
+    }
+
+    if (event.type === 'buildSuccess') {
```

Comment:
can we do it as a `switch/case` on this?
clearer to read and easier to add stuff later on.